### PR TITLE
Switch to profiles

### PR DESCRIPTION
**Pull request by ingvagabund**
Based on https://github.com/openshift/enhancements/blob/master/enhancements/scheduling/scheduler-profiles.md

### CODE REVIEW

**Comment by ingvagabund** [comment]
`schedulers.config.openshift.io` CRD needs to be updated first: https://github.com/openshift/api/pull/806

**Review by soltysh — APPROVED** [review]
/lgtm
/hold

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ func NewTargetConfigController(
 
 	configInformer.Config().V1().FeatureGates().Informer().AddEventHandler(c.eventHandler())
 	c.cachesSync = append(c.cachesSync, configInformer.Config().V1().FeatureGates().Informer().HasSynced)
+
```
Comment (damemi) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:97]: nit: new extra line

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 2]
```diff
@@ -278,10 +279,27 @@ func createTargetConfigController_v311_00_to_latest(c TargetConfigController, re
...
+		kubeSchedulerConfiguration = v410_00_assets.MustAsset("v4.1.0/config/defaultconfig-postbootstrap-lownodeutilization.yaml")
+	case v1.HighNodeUtililzation:
+		kubeSchedulerConfiguration = v410_00_assets.MustAsset("v4.1.0/config/defaultconfig-postbootstrap-highnodeutilization.yaml")
+	case v1.NoScoring:
+		kubeSchedulerConfiguration = v410_00_assets.MustAsset("v4.1.0/config/defaultconfig-postbootstrap-noscoring.yaml")
+	default:
+		return nil, false, fmt.Errorf("profile %q not recognized", config.Spec.Profile)
+	}
+
+	requiredConfigMap, _, err := resourcemerge.MergeConfigMap(configMap, "config.yaml", nil, kubeSchedulerConfiguration, operatorSpec.ObservedConfig.Raw, operatorSpec.UnsupportedConfigOverrides.Raw)
```
Comment (damemi) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:302]: how does this handle if the user sets a custom policy *and* a profile?

**Review by damemi — CHANGES_REQUESTED** [review]
the new code to use the profiles looks good, pretty straightforward. but we should probably have some sort of logging for when `policy` is still set to indicate that it's deprecated, and also disallow setting both `policy` and `profile`

**Review by damemi — APPROVED** [review]
/lgtm
/hold cancel

Thanks @ingvagabund !

**Pull request merged**